.github/workflows/main.yml

```diff
@@ -136,10 +142,35 @@ jobs:
           docker save "$CACHED_DOCKER_IMG_NAME_BASE" > /tmp/imgcache/img_base.rar
           docker image ls
 
+  build-vcxagency-redis:
+    needs:
+      - workflow-setup
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_BUILDKIT: 1
+      CACHE_KEY_REDIS: ${{needs.workflow-setup.outputs.CACHE_KEY_REDIS}}
+      CACHED_DOCKER_IMG_NAME_REDIS: ${{needs.workflow-setup.outputs.CACHED_DOCKER_IMG_NAME_REDIS}}
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+      - name: Try load from cache.
+        id: cache-image
+        uses: actions/cache@v2
+        with:
+          path: /tmp/imgcache
+          key: ${{ env.CACHE_KEY_REDIS }}
+      - name: If NOT found in cache, Build and cache image.
+        if: steps.cache-image.outputs.cache-hit != 'true'
+        run: |
+          set -x
+          docker build -t "$CACHED_DOCKER_IMG_NAME_REDIS" ci/redis
+          mkdir -p /tmp/imgcache
+          docker save "$CACHED_DOCKER_IMG_NAME_REDIS" > /tmp/imgcache/image.rar
+          docker image ls
```

Comment:
In my opinion, `set -x` and `docker ls` are useful for debugging but just clutter the code as soon when issues are resolved

Reply:
I kind of disagree about `set -x`, when things go south it's always better to have more information than less. If an intermittent problem occurs in CI, having more logs is priceless

Reply:
I do agree about `docker image ls`, I'll remove that

Reply:
Agreed